### PR TITLE
Fix/8466/messages not in order

### PR DESCRIPTION
**Pull request by osmaczko**
### What does the PR do

most likely fixes: https://github.com/status-im/status-desktop/issues/8466

- fix(chat): ensure messages ordering on model append/prepend
- feat: introduce nim tests
```bash
$ make tests-nim-linux

[Suite] empty model
  [OK] initial size

[Suite] inserting new messages
  [OK] insert same message twice
  [OK] insert in order
  [OK] insert out of order
  [OK] insert out of order (randomly)

[Suite] appending new messages
  [OK] append empty model
  [OK] append to model with only newer messages
  [OK] append to model with newer and older messages
  [OK] append to model with newer and older messages and some in between
```

### CODE REVIEW

File: ci/Jenkinsfile.tests-nim [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      name: 'VERBOSE',
+      description: 'Level of verbosity based on nimbus-build-system setup.',
+      choices: ['0', '1', '2']
+    )
+  }
+  
+  options {
+    timestamps()
+    /* Prevent Jenkins jobs from running forever */
+    timeout(time: 120, unit: 'MINUTES')
```
Comment (jrainville) [inline ci/Jenkinsfile.tests-nim:20]: Maybe a shorter timeout is in order lol

File: src/app/modules/shared_models/message_model.nim [hunk 2]
```diff
@@ -248,45 +248,30 @@ QtObject:
...
           return i
-    return 0
+    return self.items.len
 
   proc filterExistingItems(self: Model, items: seq[Item]): seq[Item] =
     for item in items:
       if(self.findIndexForMessageId(item.id) < 0):
         result &= item
 
+  proc insertItemBasedOnClock*(self: Model, item: Item)
```
Comment (jrainville) [inline src/app/modules/shared_models/message_model.nim:258]: Can you put the forward declaration at the top of the class, that's what we do usually.

**Review by jrainville — COMMENTED** [review]
Very nice. Good job on adding Nim tests
I just put two very small comments

**Review by saledjenic — APPROVED** [review]

**Review by stefandunca — APPROVED** [review]
Great to have a start for NIM unit-tests. Looking forward more to come.

**Review by jrainville — APPROVED** [review]

**Comment by anastasiyaig** [comment]
i tried to reproduce the message ordering issues and no luck . Can't guarantee it fixes the problem because it is really hard to reproduce it with clear and stable steps. However, I saw this myself with 0.8.1 and i dont see it now with this build. Fingers crossed it is fixed :D

**Review by jakubgs — COMMENTED** [review]
Looks good. But the `Check` stage could have been called `Tests`.

And this part:
```
env.PKG_URL = "${currentBuild.absoluteUrl}/consoleText"
```
Isn't really an `Upload` stage since it's not uploading anything.
Could have been put under `post` as `always` stage.

**Pull request merged**